src/main/scala/org/broadinstitute/gpp/poolq3/barcode/DmuxedBarcodeSource.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022 The Broad Institute, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package org.broadinstitute.gpp.poolq3.barcode
+
+import org.broadinstitute.gpp.poolq3.parser.{CloseableIterable, CloseableIterator, DmuxedIterable}
+import org.broadinstitute.gpp.poolq3.types.Read
+
+final class DmuxedBarcodeSource(parser: DmuxedIterable, rowPolicy: BarcodePolicy, umiPolicyOpt: Option[BarcodePolicy])
+    extends CloseableIterable[Barcodes] {
+
+  private def colBarcodeOpt = parser.indexBarcode.map(bc => FoundBarcode(bc.toCharArray, 0))
```

Comment:
Correct me if I'm wrong, but it looks like this rather expensive function will return the same value millions of times in a row (i.e. until the next file starts).  If that is indeed true, then can you find a way to cache the value?